Unify Schema Deletion Confirmation UX and Add Hard Delete Warning Modal (#2021)

Prankisster opened the pull request:
## Summary of Changes

This PR refactors the confirmation flow for schema version and subject deletions:

* Fixes #2021
* **Unified confirmation UX**: Both schema version and subject deletions now use consistent confirmation prompts based on schema name and version, rather than including deletion type (e.g., `"hard:<name>"`).
* **Introduced a modal warning step for hard deletes**: Users performing hard deletes will now receive a final modal confirmation prompt to prevent accidental destructive actions.
* **Minimal refactoring applied**: Common logic reused while preserving the original flow structure; no breaking changes introduced.

### Alternatives Considered

* Using a single `confirmDeletion()` for both versions and subjects was considered but avoided for now to maintain future extensibility and context-specific prompts
* Placing modal logic inside quick pick step was ruled out in favour of separation of concerns and clarity

---

## Any additional details or context that should be provided?

* **Before**: Hard deletes relied solely on a `QuickPickItem` prefix and lacked a strong secondary confirmation, increasing the risk of accidental irreversible deletions
* **After**: UI explicitly confirms intent with a modal before executing any hard delete

---

## Pull request checklist

Please check if your PR fulfils the following (if applicable):

##### Tests

* [ ] Added new
* [x] Updated existing (indirectly, behaviour of confirmation helpers and delete commands)
* [ ] Deleted existing

##### Other

* [x] All new disposables (event listeners, views, channels, etc.) collected as `Disposable[]` for eventual cleanup?
* [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md)?
* [x] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/docs/cli.md)?

  ```sh
  gulp clicktest
  ```

jlrobins commented on src/commands/utils/schemas.ts:
```diff
@@ -74,69 +74,48 @@ export function getSchemaDeletionValidatorAndPlaceholder(
+    }
+    return {
+      message: `Enter "${subject.name}" to confirm deletion, escape to cancel.`,
+      severity: vscode.InputBoxValidationSeverity.Error,
```
```suggestion
      severity: vscode.InputBoxValidationSeverity.Warning,
```

Prankisster commented:
Thanks for the feedback!

I've addressed the requested changes:

- [X] Fixed failing tests and adjusted expectations to align with the updated validator logic (which no longer differentiates between soft/hard confirmation strings)
- [X] Ran `gulp format` for uniform indentation style

Let me know if any further tweaks are needed — happy to refine further!

jlrobins commented:
/sem-approve

shouples reviewed (changes requested):
Looking great @Prankisster! I just have one minor cleanup suggestion, and once that's addressed I think we're good to merge this.

Prankisster commented:
Thanks for pointing that out! I've removed the explicit "Cancel" text as suggested. Let me know if there’s anything else you'd like adjusted—happy to make further changes.
